### ISSUE TITLE
Complete redesign of plugin for Kibana 6.3

### ISSUE THREAD

**Issue by martijnrondeel** [bitsensor/elastalert-kibana-plugin#56]
The past two days I've completely rewritten the plugin for Kibana 6.3. Notable changes include:

- Rewritten from Angular 1 to React.
- Using [Elastic UI](https://github.com/elastic/eui) instead of Material Design.
- Some performance improvements and quality of life changes.

I would really appreciate it if more people could test this new version and reply to this issue with any bugs or suggestions they have. The plugin is only compatible with Kibana 6.3+ and can be installed by copying the address of the file below and then installing it in your Kibana installation: `bin/kibana-plugin install URL-HERE`

Please see the [releases page](https://github.com/bitsensor/elastalert-kibana-plugin/releases) and grab the pre-release there.

~~Kibana 6.3.2: [elastalert-kibana-plugin-0.0.3.zip](https://github.com/bitsensor/elastalert-kibana-plugin/files/2235470/elastalert-kibana-plugin-0.0.3.zip)~~
~~Kibana 6.3.1: [elastalert-kibana-plugin-0.0.3.zip](https://github.com/bitsensor/elastalert-kibana-plugin/files/2235477/elastalert-kibana-plugin-0.0.3.zip)~~
~~[elastalert-kibana-plugin-0.0.2.zip](https://github.com/bitsensor/elastalert-kibana-plugin/files/2206312/elastalert-kibana-plugin-0.0.2.zip)~~
~~[elastalert-kibana-plugin-0.0.1.zip](https://github.com/bitsensor/elastalert-kibana-plugin/files/2189140/elastalert-kibana-plugin-0.0.0.zip)~~

Screenshots of the new version:

![image](https://user-images.githubusercontent.com/5266086/42641732-7ef1eab2-85f5-11e8-901c-96790258f260.png)
![image](https://user-images.githubusercontent.com/5266086/42641762-93d1cf24-85f5-11e8-89b3-e1dbc763e88c.png)
![image](https://user-images.githubusercontent.com/5266086/42641822-b4cb85d0-85f5-11e8-916f-b0e8bc279a8c.png)

**TODO:**

- [ ] YAML autocompletion and linting is not working yet in the rule editor
- [ ] Templates cannot be selected yet
- [ ] Configuration through environment variables
- [ ] Plugin installation hangs on CentOS (https://github.com/bitsensor/elastalert-kibana-plugin/issues/56#issuecomment-405455297)

**Comment by martijnrondeel** [comment]
If anyone wants to try the new plugin with Kibana 6.3 or the unreleased 6.3.2, let me know and I'll compile a version for those versions.

**Comment by k2patel** [comment]
Hi @martijnrondeel 
Is there any notes for the non dockerized install for plugin ?

**Comment by martijnrondeel** [comment]
@k2patel You mean for our ElastAlert server fork? If so, yes, you can build and run it [manually](https://github.com/bitsensor/elastalert#building-from-source).

**Comment by davidkarlsen** [comment]
@martijnrondeel Is it possible to configure elastalert.serverHost: 123.0.0.1 via env-vars instead?

**Comment by martijnrondeel** [comment]
@davidkarlsen No that is not possible currently, however I can add support for that if needed.

**Comment by davidkarlsen** [comment]
That would be great - in the meantime, what should the configuration values look like? I couldn't find docs anywhere.

**Comment by tbnjha** [comment]
@martijnrondeel 
I am stuck at the following step:
> Optimizing and caching browser bundles...

I have kibana 6.3.1 installed.
> [root@localhost ~]# /usr/share/kibana/bin/kibana version
6.3.1


The full trace of plugin installation is below:

> [root@localhost kibana]# bin/kibana-plugin install https://github.com/bitsensor/elastalert-kibana-plugin/files/2189140/elastalert-kibana-plugin-0.0.0.zip
Attempting to transfer from https://github.com/bitsensor/elastalert-kibana-plugin/files/2189140/elastalert-kibana-plugin-0.0.0.zip
Transferring 21636797 bytes....................
Transfer complete
Retrieving metadata from plugin archive
Extracting plugin archive
Extraction complete
Optimizing and caching browser bundles...

**Comment by martijnrondeel** [comment]
@tbnjha How long did you wait before canceling the installation? I remember the 'Optimizing and caching browser bundles' took quite some time on my laptop as well. Could you remove the plugin (`bin/kibana-plugin remove pluginhere`) and then install again but let it run for ~10 minutes.

@davidkarlsen The following configuration options are available:

```
elastalert.enabled (defaults to true)
elastalert.serverHost (defaults to localhost)
elastalert.serverPort (defaults to 3030)
elastalert.serverSsl (defaults to false)
```

The rewritten plugin will come with documentation of the configuration properties and a new contribution and release guide btw :)

**Comment by tbnjha** [comment]
@martijnrondeel I left it running for 25mins today, but still no luck. I was trying it on _**CentOS Linux release 7.5.1804 (Core).**_ Could it be anything related to that. Will try installing it on ubuntu 18.04 dist today, just to double confirm on the OS/Package installations.

**Comment by martijnrondeel** [comment]
@tbnjha Good to hear it works on Ubuntu. I'll investigate the CentOS issue. Thanks for testing!

**Comment by gruaig** [comment]
log   [14:57:17.082] [fatal] "elastalert.enabled", "elastalert.serverHost", and "elastalert.serverPort" settings were not applied. Check for spelling errors and ensure that expected plugins are installed.
FATAL "elastalert.enabled", "elastalert.serverHost", and "elastalert.serverPort" settings were not applied. Check for spelling errors and ensure that expected plugins are installed.
[root@elk_production-elk-kibana-1 lib]# 
@davidkarlsen 
I set 
elastalert.enabled (defaults to true)
elastalert.serverHost (defaults to localhost)
elastalert.serverPort (defaults to 3030)
elastalert.serverSsl (defaults to false)
in kibana.yml 
elastalert.enabled (defaults to true)
elastalert.serverHost: (defaults to localhost)
elastalert.serverPort: (defaults to 3030)
elastalert.serverSsl: (defaults to false)

**Comment by martijnrondeel** [comment]
@gruaig Does a line like below show up in the log?

```
[info][status][plugin:elastalert-kibana-plugin@0.0.1] Status changed from uninitialized to green - Ready
```

**Comment by martijnrondeel** [comment]
@gruaig Ah my bad. I actually accidentally changed the plugin ID which makes the configuration key `elastalert-kibana-plugin` instead of just `elastalert`. You can use the old version with `elastalert-kibana-plugin.serverHost` etc. settings or grab this new version (which also has some other improvements): [elastalert-kibana-plugin-0.0.2.zip](https://github.com/bitsensor/elastalert-kibana-plugin/files/2206312/elastalert-kibana-plugin-0.0.2.zip)

CC @davidkarlsen (your configuration probably won't work either with the first version)

**Comment by gruaig** [comment]
@martijnrondeel 
 [15:25:45.283] [warning][security] Session cookies will be transmitted over insecure connections. This is not recommended.
  log   [15:25:45.333] [info][status][plugin:grokdebugger@6.3.1] Status changed from uninitialized to yellow - Waiting for Elasticsearch
  log   [15:25:45.340] [info][status][plugin:dashboard_mode@6.3.1] Status changed from uninitialized to green - Ready
  log   [15:25:45.343] [info][status][plugin:logstash@6.3.1] Status changed from uninitialized to yellow - Waiting for Elasticsearch
  log   [15:25:45.368] [info][status][plugin:apm@6.3.1] Status changed from uninitialized to green - Ready
  log   [15:25:45.376] [info][status][plugin:console@6.3.1] Status changed from uninitialized to green - Ready
  log   [15:25:45.379] [info][status][plugin:console_extensions@6.3.1] Status changed from uninitialized to green - Ready
  log   [15:25:45.383] [info][status][plugin:metrics@6.3.1] Status changed from uninitialized to green - Ready
  log   [15:25:46.334] [warning][reporting] Generating a random key for xpack.reporting.encryptionKey. To prevent pending reports from failing on restart, please set xpack.reporting.encryptionKey in kibana.yml
  log   [15:25:46.338] [info][status][plugin:reporting@6.3.1] Status changed from uninitialized to yellow - Waiting for Elasticsearch
  log   [15:25:46.848] [info][status][plugin:elasticsearch@6.3.1] Status changed from yellow to green - Ready
  log   [15:25:46.862] [info][license][xpack] Imported license information from Elasticsearch for the [data] cluster: mode: trial | status: active | expiry date: 2018-08-06T23:59:59+00:00
  log   [15:25:46.896] [info][status][plugin:xpack_main@6.3.1] Status changed from yellow to green - Ready
  log   [15:25:46.897] [info][status][plugin:searchprofiler@6.3.1] Status changed from yellow to green - Ready
  log   [15:25:46.898] [info][status][plugin:ml@6.3.1] Status changed from yellow to green - Ready
  log   [15:25:46.899] [info][status][plugin:tilemap@6.3.1] Status changed from yellow to green - Ready
  log   [15:25:46.900] [info][status][plugin:watcher@6.3.1] Status changed from yellow to green - Ready
  log   [15:25:46.901] [info][status][plugin:index_management@6.3.1] Status changed from yellow to green - Ready
  log   [15:25:46.902] [info][status][plugin:graph@6.3.1] Status changed from yellow to green - Ready
  log   [15:25:46.903] [info][status][plugin:security@6.3.1] Status changed from yellow to green - Ready
  log   [15:25:46.904] [info][status][plugin:grokdebugger@6.3.1] Status changed from yellow to green - Ready
  log   [15:25:46.904] [info][status][plugin:logstash@6.3.1] Status changed from yellow to green - Ready
  log   [15:25:46.905] [info][status][plugin:reporting@6.3.1] Status changed from yellow to green - Ready
  log   [15:25:46.927] [info][kibana-monitoring][monitoring-ui] Starting all Kibana monitoring collectors
  log   [15:25:46.941] [info][license][xpack] Imported license information from Elasticsearch for the [monitoring] cluster: mode: trial | status: active | expiry date: 2018-08-06T23:59:59+00:00
  log   [15:26:03.931] [info][listening] Server running at http://0:5601
  log   [15:31:50.242] [info][status][plugin:kibana@6.3.1] Status changed from uninitialized to green - Ready
  log   [15:31:50.296] [info][status][plugin:elasticsearch@6.3.1] Status changed from uninitialized to yellow - Waiting for Elasticsearch
  log   [15:31:50.300] [info][status][plugin:xpack_main@6.3.1] Status changed from uninitialized to yellow - Waiting for Elasticsearch
  log   [15:31:50.306] [info][status][plugin:searchprofiler@6.3.1] Status changed from uninitialized to yellow - Waiting for Elasticsearch
  log   [15:31:50.310] [info][status][plugin:ml@6.3.1] Status changed from uninitialized to yellow - Waiting for Elasticsearch
  log   [15:31:50.357] [info][status][plugin:tilemap@6.3.1] Status changed from uninitialized to yellow - Waiting for Elasticsearch
  log   [15:31:50.360] [info][status][plugin:watcher@6.3.1] Status changed from uninitialized to yellow - Waiting for Elasticsearch
  log   [15:31:50.377] [info][status][plugin:license_management@6.3.1] Status changed from uninitialized to green - Ready
  log   [15:31:50.379] [info][status][plugin:index_management@6.3.1] Status changed from uninitialized to yellow - Waiting for Elasticsearch
  log   [15:31:50.539] [info][status][plugin:timelion@6.3.1] Status changed from uninitialized to green - Ready
  log   [15:31:50.541] [info][status][plugin:graph@6.3.1] Status changed from uninitialized to yellow - Waiting for Elasticsearch
  log   [15:31:50.545] [info][status][plugin:monitoring@6.3.1] Status changed from uninitialized to green - Ready
  log   [15:31:50.547] [info][status][plugin:security@6.3.1] Status changed from uninitialized to yellow - Waiting for Elasticsearch
  log   [15:31:50.548] [warning][security] Generating a random key for xpack.security.encryptionKey. To prevent sessions from being invalidated on restart, please set xpack.security.encryptionKey in kibana.yml
  log   [15:31:50.551] [warning][security] Session cookies will be transmitted over insecure connections. This is not recommended.
  log   [15:31:50.589] [info][status][plugin:grokdebugger@6.3.1] Status changed from uninitialized to yellow - Waiting for Elasticsearch
  log   [15:31:50.600] [info][status][plugin:dashboard_mode@6.3.1] Status changed from uninitialized to green - Ready
  log   [15:31:50.604] [info][status][plugin:logstash@6.3.1] Status changed from uninitialized to yellow - Waiting for Elasticsearch
  log   [15:31:50.635] [info][status][plugin:apm@6.3.1] Status changed from uninitialized to green - Ready
  log   [15:31:50.645] [info][status][plugin:console@6.3.1] Status changed from uninitialized to green - Ready
  log   [15:31:50.649] [info][status][plugin:console_extensions@6.3.1] Status changed from uninitialized to green - Ready
  log   [15:31:50.656] [info][status][plugin:elastalert-kibana-plugin@0.0.0] Status changed from uninitialized to green - Ready
  log   [15:31:50.661] [info][status][plugin:metrics@6.3.1] Status changed from uninitialized to green - Ready
  log   [15:31:51.524] [warning][reporting] Generating a random key for xpack.reporting.encryptionKey. To prevent pending reports from failing on restart, please set xpack.reporting.encryptionKey in kibana.yml
  log   [15:31:51.529] [info][status][plugin:reporting@6.3.1] Status changed from uninitialized to yellow - Waiting for Elasticsearch
  log   [15:31:51.728] [info][listening] Server running at http://0:5601
  log   [15:31:51.750] [info][status][plugin:elasticsearch@6.3.1] Status changed from yellow to green - Ready
  log   [15:31:51.761] [info][license][xpack] Imported license information from Elasticsearch for the [data] cluster: mode: trial | status: active | expiry date: 2018-08-06T23:59:59+00:00
  log   [15:31:51.802] [info][status][plugin:xpack_main@6.3.1] Status changed from yellow to green - Ready
  log   [15:31:51.802] [info][status][plugin:searchprofiler@6.3.1] Status changed from yellow to green - Ready
  log   [15:31:51.803] [info][status][plugin:ml@6.3.1] Status changed from yellow to green - Ready
  log   [15:31:51.804] [info][status][plugin:tilemap@6.3.1] Status changed from yellow to green - Ready
  log   [15:31:51.804] [info][status][plugin:watcher@6.3.1] Status changed from yellow to green - Ready
  log   [15:31:51.805] [info][status][plugin:index_management@6.3.1] Status changed from yellow to green - Ready
  log   [15:31:51.805] [info][status][plugin:graph@6.3.1] Status changed from yellow to green - Ready
  log   [15:31:51.806] [info][status][plugin:security@6.3.1] Status changed from yellow to green - Ready
  log   [15:31:51.807] [info][status][plugin:grokdebugger@6.3.1] Status changed from yellow to green - Ready
  log   [15:31:51.807] [info][status][plugin:logstash@6.3.1] Status changed from yellow to green - Ready
  log   [15:31:51.808] [info][status][plugin:reporting@6.3.1] Status changed from yellow to green - Ready
  log   [15:31:51.818] [info][kibana-monitoring][monitoring-ui] Starting all Kibana monitoring collectors
  log   [15:31:51.829] [info][license][xpack] Imported license information from Elasticsearch for the [monitoring] cluster: mode: trial | status: active | expiry date: 2018-08-06T23:59:59+00:00

**Comment by martijnrondeel** [comment]
@gruaig The plugin should be available:

> log [15:31:50.656] [info][status][plugin:elastalert-kibana-plugin@0.0.0] Status changed from uninitialized to green - Ready

Does it show up in the sidebar?

**Comment by gruaig** [comment]
@martijnrondeel   log   [15:44:59.877] [fatal] "elastalert.serverHost" setting was not applied. Check for spelling errors and ensure that expected plugins are installed.
FATAL "elastalert.serverHost" setting was not applied. Check for spelling errors and ensure that expected plugins are installed.
Still not starting. The previous was with the settings removed.

**Comment by martijnrondeel** [comment]
Hmm ok. I’ll try some things and come back to you when the configuration
issue is fixed. Thanks for testing!

**Comment by martijnrondeel** [comment]
@gruaig Ah ok. Let me know how it feels. I’ll fix it so the configuration key is `elastalert` again.

**Comment by martijnrondeel** [comment]
Hmm ok, thanks for testing @gruaig. I’ll do some testing on our machines and fix the issue!

**Comment by davidkarlsen** [comment]
Same here, see: https://hub.docker.com/r/evryfs/kibana-docker/builds/bgfsagxwmm7g9vyjguwlzm/

**Comment by gruaig** [comment]
Eh Volia. 
![screenshot from 2018-07-19 09-04-15](https://user-images.githubusercontent.com/5451282/42929747-dcdff082-8b32-11e8-9218-5c3645051ada.png)
{
  "name": "elastalert-kibana-plugin",
  "version": "0.0.2",
  "description": "Create, test and edit ElastAlert rules within Kibana.",
  "main": "index.js",
  "license": "SEE LICENSE IN LICENSE.md",
  "kibana": {
    "version": "6.3.1",
    "templateVersion": "1.0.0"
  },
  "dependencies": {
    "@elastic/eui": "^3.0.0"
  },
  "build": {
    "git": {},
    "date": "Wed Jul 18 2018 17:15:14 GMT+0200 (CEST)"
  }
}

**Comment by martijnrondeel** [comment]
@gruaig Ah that's great. What was the connection issue?

**Comment by gruaig** [comment]
@martijnrondeel  It was to do with my host network. Can you upload your branch please I want to fork it.

**Comment by gruaig** [comment]
@martijnrondeel 

New bug. I've disabled X-pack and installed the plugin.
bash-4.2$ ./kibana-plugin lis

 ERROR  unknown command lis

  Usage: bin/kibana-plugin [command] [options]
  
  The Kibana plugin manager enables you to install and remove plugins that provide additional functionality to Kibana
  
  Commands:
    list  [options]                 list installed plugins
    install  [options] <plugin/url> install a plugin
    remove  [options] <plugin>      remove a plugin
    help  <command>                 get the help for a specific command

bash-4.2$ ./kibana-plugin list
elastalert-kibana-plugin@0.0.2

But it no longer appears in in the menu and it is not loaded when Kib starts up

**Comment by martijnrondeel** [comment]
@gruaig Did you install the plugin before or after disabling x-pack?

**Comment by gruaig** [comment]
Hey I've gotten around it. I installed the plugin then removed x-pack it works then. Next on I'm not convinced that elastalert is working with 6.3.x. Even basic alerts will return errors like,

`elastalert_error - {'message': "Error running query: TransportError(400, u'parsing_exception', u'no [query] registered for [filtered]')", 'traceback': ['Traceback (most recent call last):', '  File "elastalert/elastalert.py", line 369, in get_hits', '    **extra_args', '  File "build/bdist.linux-x86_64/egg/elasticsearch/client/utils.py", line 73, in _wrapped', '    return func(*args, params=params, **kwargs)', '  File "build/bdist.linux-x86_64/egg/elasticsearch/client/__init__.py", line 623, in search', "    doc_type, '_search'), params=params, body=body)", '  File "build/bdist.linux-x86_64/egg/elasticsearch/transport.py", line 312, in perform_request', '    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)', '  File "build/bdist.linux-x86_64/egg/elasticsearch/connection/http_requests.py", line 90, in perform_request', '    self._raise_error(response.status_code, raw_data)', '  File "build/bdist.linux-x86_64/egg/elasticsearch/connection/base.py", line 125, in _raise_error', '    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)', "RequestError: TransportError(400, u'parsing_exception', u'no [query] registered for [filtered]')"], 'data': {'query': {'sort': [{'@timestamp': {'order': 'asc'}}], 'query': {'filtered': {'filter': {'bool': {'must': [{'range': {'@timestamp': {'gt': '2018-07-18T15:15:43.887874Z', 'lte': '2018-07-18T15:16:43.887874Z'}}}, {'query': {'query_string': {'query': 'ERROR'}}}]}}}}}, 'rule': 'Example rule'}}` @martijnrondeel Wouldn't imagine it's anything to do with your plugin.
 
Example of the rule
`es_host: 10.10.209.51
es_port: 9200
name: Example rule
type: frequency
index: metric-*
num_events: 50
timeframe:
    hours: 4
filter:
- query:
    query_string:
      query: "ERROR"
alert:
- "email"
email:
- "elastalert@example.com"
`
Do you have any working rules I may try?

**Comment by martijnrondeel** [comment]
@gruaig I just published a new build of ElastAlert on Docker Hub. Can you `docker pull bitsensor/elastalert:latest` run it, and tell me if the rule works?

**Comment by gruaig** [comment]
@martijnrondeel 
On startup
`:03:18.544Z  INFO elastalert-server: Config:  No config.dev.json file was found in /opt/elastalert-server/config/config.dev.json.
16:03:18.546Z  INFO elastalert-server: Config:  Proceeding to look for normal config file.
16:03:18.546Z  INFO elastalert-server: Config:  A config file was found in /opt/elastalert-server/config/config.json. Using that config.
16:03:18.565Z  INFO elastalert-server: Router:  Listening for GET request on /.
16:03:18.565Z  INFO elastalert-server: Router:  Listening for GET request on /status.
16:03:18.565Z  INFO elastalert-server: Router:  Listening for GET request on /status/control/:action.
16:03:18.565Z  INFO elastalert-server: Router:  Listening for GET request on /status/errors.
16:03:18.565Z  INFO elastalert-server: Router:  Listening for GET request on /rules.
16:03:18.567Z  INFO elastalert-server: Router:  Listening for GET request on /rules/:id.
16:03:18.567Z  INFO elastalert-server: Router:  Listening for POST request on /rules/:id.
16:03:18.568Z  INFO elastalert-server: Router:  Listening for DELETE request on /rules/:id.
16:03:18.568Z  INFO elastalert-server: Router:  Listening for GET request on /templates.
16:03:18.568Z  INFO elastalert-server: Router:  Listening for GET request on /templates/:id.
16:03:18.568Z  INFO elastalert-server: Router:  Listening for POST request on /templates/:id.
16:03:18.568Z  INFO elastalert-server: Router:  Listening for DELETE request on /templates/:id.
16:03:18.568Z  INFO elastalert-server: Router:  Listening for POST request on /test.
16:03:18.568Z  INFO elastalert-server: Router:  Listening for GET request on /config.
16:03:18.568Z  INFO elastalert-server: Router:  Listening for POST request on /config.
16:03:18.568Z  INFO elastalert-server: Router:  Listening for POST request on /download.
16:03:18.573Z  INFO elastalert-server: ProcessController:  Starting ElastAlert
16:03:18.573Z  INFO elastalert-server: ProcessController:  Creating index
16:03:19.083Z  INFO elastalert-server:
    ProcessController:  Elastic Version:6
    Mapping used for string:{'type': 'keyword'}
    Index elastalert_status already exists. Skipping index creation.
    
16:03:19.084Z  INFO elastalert-server: ProcessController:  Index create exited with code 0
16:03:19.085Z  INFO elastalert-server: ProcessController:  Starting elastalert with arguments [none]
16:03:19.097Z  INFO elastalert-server: ProcessController:  Started Elastalert (PID: 35)
16:03:19.100Z  INFO elastalert-server: Server:  Server listening on port 3030
16:03:19.100Z  INFO elastalert-server: Server:  Server started
16:03:19.797Z ERROR elastalert-server:
    ProcessController:  WARNING:elasticsearch:GET http://elasticsearch:9200/elastalert_status/elastalert/_search?size=1000 [status:400 request:0.015s]
    
16:03:19.798Z ERROR elastalert-server:
    ProcessController:  ERROR:root:Error finding recent pending alerts: TransportError(400, u'search_phase_execution_exception', u'No mapping found for [alert_time] in order to sort on') {'sort': {'alert_time': {'order': 'asc'}}, 'query': {'bool': {'filter': {'range': {'alert_time': {'to': '2018-07-19T16:03:19.775209Z', 'from': '2018-07-17T16:03:19.775189Z'}}}, 'must': {'query_string': {'query': '!_exists_:aggregate_id AND alert_sent:false'}}}}}
    Traceback (most recent call last):
      File "/opt/elastalert/elastalert/elastalert.py", line 1500, in find_recent_pending_alerts
        size=1000)
      File "/usr/lib/python2.7/site-packages/elasticsearch-6.3.0-py2.7.egg/elasticsearch/client/utils.py", line 76, in _wrapped
        return func(*args, params=params, **kwargs)
      File "/usr/lib/python2.7/site-packages/elasticsearch-6.3.0-py2.7.egg/elasticsearch/client/__init__.py", line 655, in search
        doc_type, '_search'), params=params, body=body)
      File "/usr/lib/python2.7/site-packages/elasticsearch-6.3.0-py2.7.egg/elasticsearch/transport.py", line 318, in perform_request
        status, headers_response, data = connection.perform_request(method, url, params, body, headers=headers, ignore=ignore, timeout=timeout)
      File "/usr/lib/python2.7/site-packages/elasticsearch-6.3.0-py2.7.egg/elasticsearch/connection/http_requests.py", line 90, in perform_request
        self._raise_error(response.status_code, raw_data)
      File "/usr/lib/python2.7/site-packages/elasticsearch-6.3.0-py2.7.egg/elasticsearch/connection/base.py", line 125, in _raise_error
        raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
    RequestError: TransportError(400, u'search_phase_execution_exception', u'No mapping found for [alert_time] in order to sort on')
    
16:03:24.793Z ERROR elastalert-server:
    ProcessController:  WARNING:elasticsearch:GET http://elasticsearch:9200/elastalert_status/elastalert/_search?size=1000 [status:400 request:0.012s]
    
16:03:24.794Z ERROR elastalert-server:
    ProcessController:  ERROR:root:Error finding recent pending alerts: TransportError(400, u'search_phase_execution_exception', u'No mapping found for [alert_time] in order to sort on') {'sort': {'alert_time': {'order': 'asc'}}, 'query': {'bool': {'filter': {'range': {'alert_time': {'to': '2018-07-19T16:03:24.780455Z', 'from': '2018-07-17T16:03:24.780395Z'}}}, 'must': {'query_string': {'query': '!_exists_:aggregate_id AND alert_sent:false'}}}}}
    Traceback (most recent call last):
      File "/opt/elastalert/elastalert/elastalert.py", line 1500, in find_recent_pending_alerts
        size=1000)
      File "/usr/lib/python2.7/site-packages/elasticsearch-6.3.0-py2.7.egg/elasticsearch/client/utils.py", line 76, in _wrapped
        return func(*args, params=params, **kwargs)
      File "/usr/lib/python2.7/site-packages/elasticsearch-6.3.0-py2.7.egg/elasticsearch/client/__init__.py", line 655, in search
        doc_type, '_search'), params=params, body=body)
      File "/usr/lib/python2.7/site-packages/elasticsearch-6.3.0-py2.7.egg/elasticsearch/transport.py", line 318, in perform_request
        status, headers_response, data = connection.perform_request(method, url, params, body, headers=headers, ignore=ignore, timeout=timeout)
      File "/usr/lib/python2.7/site-packages/elasticsearch-6.3.0-py2.7.egg/elasticsearch/connection/http_requests.py", line 90, in perform_request
        self._raise_error(response.status_code, raw_data)
      File "/usr/lib/python2.7/site-packages/elasticsearch-6.3.0-py2.7.egg/elasticsearch/connection/base.py", line 125, in _raise_error
        raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
    RequestError: TransportError(400, u'search_phase_execution_exception', u'No mapping found for [alert_time] in order to sort on')
    
16:03:29.800Z ERROR elastalert-server:
    ProcessController:  WARNING:elasticsearch:GET http://elasticsearch:9200/elastalert_status/elastalert/_search?size=1000 [status:400 request:0.014s]
    ERROR:root:Error finding recent pending alerts: TransportError(400, u'search_phase_execution_exception', u'No mapping found for [alert_time] in order to sort on') {'sort': {'alert_time': {'order': 'asc'}}, 'query': {'bool': {'filter': {'range': {'alert_time': {'to': '2018-07-19T16:03:29.785603Z', 'from': '2018-07-17T16:03:29.785526Z'}}}, 'must': {'query_string': {'query': '!_exists_:aggregate_id AND alert_sent:false'}}}}}
    Traceback (most recent call last):
      File "/opt/elastalert/elastalert/elastalert.py", line 1500, in find_recent_pending_alerts
        size=1000)
      File "/usr/lib/python2.7/site-packages/elasticsearch-6.3.0-py2.7.egg/elasticsearch/client/utils.py", line 76, in _wrapped
        return func(*args, params=params, **kwargs)
      File "/usr/lib/python2.7/site-packages/elasticsearch-6.3.0-py2.7.egg/elasticsearch/client/__init__.py", line 655, in search
        doc_type, '_search'), params=params, body=body)
      File "/usr/lib/python2.7/site-packages/elasticsearch-6.3.0-py2.7.egg/elasticsearch/transport.py", line 318, in perform_request
        status, headers_response, data = connection.perform_request(method, url, params, body, headers=headers, ignore=ignore, timeout=timeout)
      File "/usr/lib/python2.7/site-packages/elasticsearch-6.3.0-py2.7.egg/elasticsearch/connection/http_requests.py", line 90, in perform_request
        self._raise_error(response.status_code, raw_data)
      File "/usr/lib/python2.7/site-packages/elasticsearch-6.3.0-py2.7.egg/elasticsearch/connection/base.py", line 125, in _raise_error
        raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
    RequestError: TransportError(400, u'search_phase_execution_exception', u'No mapping found for [alert_time] in order to sort on')

`
On rule send 
`
# Alert when the rate of events exceeds a threshold

# (Optional)
# Elasticsearch host
# es_host: elasticsearch.example.com

# (Optional)
# Elasticsearch port
# es_port: 14900

# (OptionaL) Connect with SSL to Elasticsearch
#use_ssl: True

# (Optional) basic-auth username and password for Elasticsearch
#es_username: someusername
#es_password: somepassword

# (Required)
# Rule name, must be unique
name: Example frequency rule

# (Required)
# Type of alert.
# the frequency rule type alerts when num_events events occur with timeframe time
type: frequency

# (Required)
# Index to search, wildcard supported
index: logstash-*

# (Required, frequency specific)
# Alert when this many documents matching the query occur within a timeframe
num_events: 50

# (Required, frequency specific)
# num_events must occur within this amount of time to trigger an alert
timeframe:
  hours: 4

# (Required)
# A list of Elasticsearch filters used for find events
# These filters are joined with AND and nested in a filtered query
# For more info: http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/query-dsl.html
filter:
- term:
    some_field: "some_value"

# (Required)
# The alert is use when a match is found
alert:
- "email"

# (required, email specific)
# a list of email addresses to send alerts to
email:
- "elastalert@example.com"
`

Console output
Successfully loaded Example frequency rule



Would have written the following documents to writeback index (default is elastalert_status):

elastalert_status - {'hits': 0, 'matches': 0, '@timestamp': datetime.datetime(2018, 7, 19, 16, 6, 27, 707663, tzinfo=tzutc()), 'rule_name': 'Example frequency rule', 'starttime': datetime.datetime(2018, 7, 18, 16, 6, 22, 241856, tzinfo=tzutc()), 'endtime': datetime.datetime(2018, 7, 19, 16, 6, 22, 241856, tzinfo=tzutc()), 'time_taken': 5.4556519985198975}

`
No alert has been sent yet though

**Comment by martijnrondeel** [comment]
Hmm that is weird @gruaig. After creating an index pattern in Kibana for `elastalert_*` I'm able to receive slack alerts without any errors from the following test rule:

```
name: Rule test
type: frequency
index: elastalert_*
num_events: 1
timeframe:
    hours: 4
filter: []

# The alert is use when a match is found
alert:
  - slack
  
alert_subject: " ElastAlert"
alert_text: "Testing new ElastAlert"
alert_text_type: alert_text_only

slack_msg_color: warning
slack_webhook_url: "redacted"
slack_username_override: "ElastAlert"
```

**Comment by gruaig** [comment]
All good. Got it all sorted thanks. No problems yet. If you need any help
testing let me know. Will you be making the repo public

**Comment by martijnrondeel** [comment]
Yep my plan is to push the new version to the develop branch, do a GitHub pre-release of it and then once it's stable enough merge to master and push a final release.

**Comment by kavin2606** [comment]
Hi Martijnrondeel,

I am new to Elastalert. I want to create a frequency email alert depending on the syslogs in ES. I'm new to this it would be great if you could help me out.
This is the rule I tested 

# From example_rules/example_frequency.yaml
es_host: logstash
es_port: 9200
name: Example rule
type: frequency
index: logstash-*
num_events: 50
timeframe:
    hours: 1
alert:
- "email"
email:
- "iamkavinarasu@gmail.com"

And I got the following

Console output
WARNING:elasticsearch:GET http://logstash:9200/ [status:N/A request:0.003s]
Traceback (most recent call last):
  File "build/bdist.linux-x86_64/egg/elasticsearch/connection/http_requests.py", line 76, in perform_request
    response = self.session.send(prepared_request, **send_kwargs)
  File "/usr/lib/python2.7/site-packages/requests-2.18.4-py2.7.egg/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests-2.18.4-py2.7.egg/requests/adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPConnectionPool(host='logstash', port=9200): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fadbcc1cb10>: Failed to establish a new connection: [Errno -2] Name does not resolve',))

WARNING:elasticsearch:GET http://logstash:9200/ [status:N/A request:0.002s]
Traceback (most recent call last):
  File "build/bdist.linux-x86_64/egg/elasticsearch/connection/http_requests.py", line 76, in perform_request
    response = self.session.send(prepared_request, **send_kwargs)
  File "/usr/lib/python2.7/site-packages/requests-2.18.4-py2.7.egg/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests-2.18.4-py2.7.egg/requests/adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPConnectionPool(host='logstash', port=9200): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fadbcc52850>: Failed to establish a new connection: [Errno -2] Name does not resolve',))

WARNING:elasticsearch:GET http://logstash:9200/ [status:N/A request:0.002s]
Traceback (most recent call last):
  File "build/bdist.linux-x86_64/egg/elasticsearch/connection/http_requests.py", line 76, in perform_request
    response = self.session.send(prepared_request, **send_kwargs)
  File "/usr/lib/python2.7/site-packages/requests-2.18.4-py2.7.egg/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests-2.18.4-py2.7.egg/requests/adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPConnectionPool(host='logstash', port=9200): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fadbcc52910>: Failed to establish a new connection: [Errno -2] Name does not resolve',))

WARNING:elasticsearch:GET http://logstash:9200/ [status:N/A request:0.002s]
Traceback (most recent call last):
  File "build/bdist.linux-x86_64/egg/elasticsearch/connection/http_requests.py", line 76, in perform_request
    response = self.session.send(prepared_request, **send_kwargs)
  File "/usr/lib/python2.7/site-packages/requests-2.18.4-py2.7.egg/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests-2.18.4-py2.7.egg/requests/adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPConnectionPool(host='logstash', port=9200): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fadbcc527d0>: Failed to establish a new connection: [Errno -2] Name does not resolve',))

Error connecting to ElasticSearch:
ConnectionError('N/A', "HTTPConnectionPool(host='logstash', port=9200): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fadbcc527d0>: Failed to establish a new connection: [Errno -2] Name does not resolve',))", ConnectionError(MaxRetryError("HTTPConnectionPool(host='logstash', port=9200): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fadbcc527d0>: Failed to establish a new connection: [Errno -2] Name does not resolve',))",),))

INFO:elastalert:Note: In debug mode, alerts will be logged to console but NOT actually sent.
                To send them but remain verbose, use --verbose instead.

WARNING:elasticsearch:GET http://logstash:9200/ [status:N/A request:0.002s]
Traceback (most recent call last):
  File "build/bdist.linux-x86_64/egg/elasticsearch/connection/http_requests.py", line 76, in perform_request
    response = self.session.send(prepared_request, **send_kwargs)
  File "/usr/lib/python2.7/site-packages/requests-2.18.4-py2.7.egg/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests-2.18.4-py2.7.egg/requests/adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPConnectionPool(host='logstash', port=9200): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fadbcc01450>: Failed to establish a new connection: [Errno -2] Name does not resolve',))

WARNING:elasticsearch:GET http://logstash:9200/ [status:N/A request:2.505s]
Traceback (most recent call last):
  File "build/bdist.linux-x86_64/egg/elasticsearch/connection/http_requests.py", line 76, in perform_request
    response = self.session.send(prepared_request, **send_kwargs)
  File "/usr/lib/python2.7/site-packages/requests-2.18.4-py2.7.egg/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests-2.18.4-py2.7.egg/requests/adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPConnectionPool(host='logstash', port=9200): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fadbcc01650>: Failed to establish a new connection: [Errno -2] Name does not resolve',))

WARNING:elasticsearch:GET http://logstash:9200/ [status:N/A request:0.003s]
Traceback (most recent call last):
  File "build/bdist.linux-x86_64/egg/elasticsearch/connection/http_requests.py", line 76, in perform_request
    response = self.session.send(prepared_request, **send_kwargs)
  File "/usr/lib/python2.7/site-packages/requests-2.18.4-py2.7.egg/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests-2.18.4-py2.7.egg/requests/adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPConnectionPool(host='logstash', port=9200): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fadbcc013d0>: Failed to establish a new connection: [Errno -2] Name does not resolve',))

WARNING:elasticsearch:GET http://logstash:9200/ [status:N/A request:0.003s]
Traceback (most recent call last):
  File "build/bdist.linux-x86_64/egg/elasticsearch/connection/http_requests.py", line 76, in perform_request
    response = self.session.send(prepared_request, **send_kwargs)
  File "/usr/lib/python2.7/site-packages/requests-2.18.4-py2.7.egg/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests-2.18.4-py2.7.egg/requests/adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPConnectionPool(host='logstash', port=9200): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fadbcc01190>: Failed to establish a new connection: [Errno -2] Name does not resolve',))

WARNING:elastalert:Error connecting to Elasticsearch for rule Example rule. The rule has been disabled.

ERROR:root:Error connecting to SMTP host: {}

INFO:elastalert:Skipping writing to ES: {'message': 'Error connecting to SMTP host: {}', 'traceback': ['Traceback (most recent call last):', '  File "elastalert/elastalert.py", line 1734, in send_notification_email', '    smtp.sendmail(self.from_addr, recipients, email.as_string())', '  File "/usr/lib/python2.7/smtplib.py", line 748, in sendmail', '    raise SMTPRecipientsRefused(senderrs)', 'SMTPRecipientsRefused: {}'], 'data': {'email_body': '\n\nThe rule Example rule has raised an uncaught exception.\n\nTraceback (most recent call last):\n  File "elastalert/elastalert.py", line 859, in init_rule\n    self.modify_rule_for_ES5(new_rule)\n  File "elastalert/elastalert.py", line 918, in modify_rule_for_ES5\n    if rule_es.info()[\'version\'][\'number\'].startswith(\'5\'):\n  File "build/bdist.linux-x86_64/egg/elasticsearch/client/utils.py", line 73, in _wrapped\n    return func(*args, params=params, **kwargs)\n  File "build/bdist.linux-x86_64/egg/elasticsearch/client/__init__.py", line 220, in info\n    return self.transport.perform_request(\'GET\', \'/\', params=params)\n  File "build/bdist.linux-x86_64/egg/elasticsearch/transport.py", line 312, in perform_request\n    status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)\n  File "build/bdist.linux-x86_64/egg/elasticsearch/connection/http_requests.py", line 85, in perform_request\n    raise ConnectionError(\'N/A\', str(e), e)\nConnectionError: ConnectionError(HTTPConnectionPool(host=\'logstash\', port=9200): Max retries exceeded with url: / (Caused by NewConnectionError(\'<urllib3.connection.HTTPConnection object at 0x7fadbcc01190>: Failed to establish a new connection: [Errno -2] Name does not resolve\',))) caused by: ConnectionError(HTTPConnectionPool(host=\'logstash\', port=9200): Max retries exceeded with url: / (Caused by NewConnectionError(\'<urllib3.connection.HTTPConnection object at 0x7fadbcc01190>: Failed to establish a new connection: [Errno -2] Name does not resolve\',)))\n'}}

Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main

    "__main__", fname, loader, pkg_name)

  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code

    exec code in run_globals
  File "/opt/elastalert/elastalert/test_rule.py", line 378, in <module>

    main()
  File "/opt/elastalert/elastalert/test_rule.py", line 374, in main

    test_instance.run_rule_test()
  File "/opt/elastalert/elastalert/test_rule.py", line 369, in run_rule_test

    self.run_elastalert(rule_yaml, conf, args)
  File "/opt/elastalert/elastalert/test_rule.py", line 265, in run_elastalert

    client.run_rule(rule, endtime, starttime)
  File "elastalert/elastalert.py", line 755, in run_rule

    for x in range(len(rule['agg_matches'])):
KeyError: 'agg_matches'

Thank you

**Comment by martijnrondeel** [comment]
@kavin2606 Please create a new issue for your problem. This issue is meant for feedback on the plugin rewrite.

**Comment by gruaig** [comment]
It's his host settings I would imagine. It can't resolve host=logstash.

**Comment by billyJacob** [comment]
Hi @gruaig 
I installed elastalert-kibana-plugin 0.0.2 and I have Kibana version 6.3.1 
The installation was successful but In Kibana I am not able to see a tab for ElastAlert.
Is there any config changes to be made In Kibana . If yes please let me know.

**Comment by martijnrondeel** [comment]
@billyJacob I've just uploaded a new version, 0.0.3, available to test, please check the original post (https://github.com/bitsensor/elastalert-kibana-plugin/issues/56#issue-340682166) to download it. If the plugin is not showing up please share the full log of Kibana here.

**Comment by jbprt** [comment]
I managed to install the plugin elastalert-kibana-plugin-0.0.3.zip but once I run kibana, I click on the tab elsatalert, which launches the next page. How can I find the right gateway?
Thanks
![capture](https://user-images.githubusercontent.com/41793187/43324898-878c3616-91b5-11e8-8d1e-5ca6b9a6292a.PNG)

**Comment by billyJacob** [comment]
Hi even I am facing the same issue now.

**Comment by billyJacob** [comment]
Hi @martijnrondeel

I managed to install the plugin elastalert-kibana-plugin-0.0.2.zip.
The mistake was with the installation path. I had installed it earlier in a wrong path.
Now I am facing this 502 - Bad Gateway issue.

**Comment by jbprt** [comment]
@gruaig How did you do ?

**Comment by billiaz** [comment]
I installed version 0.0.3 on my Kubernetes cluster! and going to test it tonight... I will post results.

Afterwards it will be deployed in 3-4 clusters for end usage and further testing

**Comment by billyJacob** [comment]
Hi @martijnrondeel
I installed elastalert-kibana-plugin 0.0.2 and I have Kibana version 6.3.1
When I click the the elastAlert tab Iam getting the following error 


![error](https://user-images.githubusercontent.com/41776566/43516537-770e154c-95a3-11e8-983c-04ce4b3c3880.PNG)

and in the kibana logs Iam getting the following log
{"type":"response","@timestamp":"2018-08-01T10:22:17Z","tags":[],"pid":9584,"method":"get","statusCode":502,"req":{"url":"/api/elastalert/rules","method":"get","headers":{"host":"localhost:5601","connection":"keep-alive","accept":"application/json, text/plain, */*","kbn-version":"6.3.1","user-agent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36","referer":"http://localhost:5601/app/elastalert-kibana-plugin","accept-encoding":"gzip, deflate, br","accept-language":"en-US,en;q=0.9"},"remoteAddress":"127.0.0.1","userAgent":"127.0.0.1","referer":"http://localhost:5601/app/elastalert-kibana-plugin"},"res":{"statusCode":502,"responseTime":1014,"contentLength":9},"message":"GET /api/elastalert/rules 502 1014ms - 9.0B"}

Is there any config changes to be made In Kibana . If yes please let me know.

**Comment by gruaig** [comment]
You are not running the elastalert service. I run mine in docker like so.

`sudo docker run -itd -p 3030:3030     -v `pwd`/config/elastalert.yaml:/opt/elastalert/config.yaml     -v `pwd`/config/config.json:/opt/elastalert-server/config/config.json     -v `pwd`/rules:/opt/elastalert/rules     -v `pwd`/rule_templates:/opt/elastalert/rule_templates     --net=host     --name elastalert bitsensor/elastalert:latest
Make sure your elastic then points to the machine
`

**Comment by jbprt** [comment]
We must have a docker to run elastalert?

**Comment by gruaig** [comment]
Yes you must be running elastalert. The plugin does not do this. 
https://github.com/Yelp/elastalert. How to run.

**Comment by jbprt** [comment]
ok, i must run elastalert with python?

**Comment by hgari** [comment]
Hi, i am newbie ELK user. install elastalert and elastalert kibana plugin. Install is ok, restart kibana. 
Then enter on ElastAlert option show gateway error.
Regards 
![kibana](https://user-images.githubusercontent.com/10588045/43681125-96455bde-9821-11e8-9fba-9797af9dd0dc.png)

**Comment by davidkarlsen** [comment]
@martijnrondeel please have a look at https://hub.docker.com/r/evryfs/kibana-docker/builds/bdkvpaf6pnnyodzjmgjouxq/ - it still does not like the configuration variable.

Do you have any plans to update to 6.3.2?

**Comment by jbprt** [comment]
We work on windows

**Comment by billiaz** [comment]
Rull testing Is not working.
For some reasn stucks there for ever....

<img width="884" alt="screen shot 2018-08-08 at 07 41 04" src="https://user-images.githubusercontent.com/4048920/43818371-9330999e-9ade-11e8-8bc0-91d241a23992.png">

I run it in kubernetes cluster with 
elastalert: 
elastalert-b7b8c9486-drbdd elastalert-api 05:41:02.745Z  INFO elastalert-server: Routes:  Successfully handled GET request for '/rules/:id'.

No errors at kibana, just the get request of the elastalert/test url

**Comment by scrat98** [comment]
1) In toast.js you should you <EuiPortal> wrapper like this(https://github.com/elastic/kibana/pull/17377/files) for correct work DomPortal and show toasts over the EuiOverlayMask
2) You should rewrite saving function and check schemaOnly, because of elastalert-server start elastalert pyhton module only once (https://github.com/bitsensor/elastalert/blob/46ad3dcb4c24b73ec3f16e95c056d2bf9b03211a/src/controllers/process/index.js#L90) and if elastalert python module find incorrect rule-file in rule folder it will crush and never restart. So on the client side we just can test it with schemaOnly option, but it problem on the elastalert-server side

**Comment by martijnrondeel** [comment]
@jbprt @billyJacob Please make sure you are running our [ElastAlert](https://github.com/bitsensor/elastalert) fork server as well. By default the plugin will connect to `localhost:3030`. If your ElastAlert server is running on a different host or port add/change the following options in your `config/kibana.yml` file: 

```
elastalert-kibana-plugin.serverHost: 123.0.0.1
elastalert-kibana-plugin.serverPort: 9000
```

@davidkarlsen Please check the first post, it has a link to a version for 6.3.2 and 6.3.1. Sadly I can't change the configuration variable to `elastalert` so you'll have to use `elastalert-kibana-plugin` (see above).

@billiaz Would you mind sharing the rule configuration?

@scrat98 Thanks for the advice, I'll add it to the todo list :+1:.

**Comment by martijnrondeel** [comment]
The source code of the rewritten plugin is now available in the [develop](https://github.com/bitsensor/elastalert-kibana-plugin/tree/develop) branch.

A pre-release is available on the GitHub [releases page](https://github.com/bitsensor/elastalert-kibana-plugin/releases).

**Comment by martijnrondeel** [comment]
I've published a new pre-release of the plugin for Kibana 6.4.0 [here](https://github.com/bitsensor/elastalert-kibana-plugin/releases). Please try it out and send us feedback :+1:.

**Comment by martijnrondeel** [comment]
I've pushed the rewritten plugin to master and created a [1.0.0 release](https://github.com/bitsensor/elastalert-kibana-plugin/releases/tag/1.0.0). Features like template support will be worked on now.

**Comment by meriem-ux** [comment]
hi pls i have issue with elastalert kibana plugin for kibana 7.3.2 the rule can't saved
any help plz

**Comment by aj1991** [comment]
Any ideas of how to Install/Configure ELastalert Kibana plugin on ELasticsearch version 7.x ?
